src/main/java/macrobase/analysis/stats/RandomProjection.java

```diff
@@ -0,0 +1,62 @@
+package macrobase.analysis.stats;
+
+import macrobase.analysis.pipeline.operator.MBStream;
+import macrobase.analysis.transform.FeatureTransform;
+import macrobase.conf.MacroBaseConf;
+import macrobase.conf.MacroBaseDefaults;
+import macrobase.datamodel.Datum;
+import org.apache.commons.math3.distribution.MultivariateNormalDistribution;
+import org.apache.commons.math3.linear.*;
+
+import java.util.List;
+
+
+public class RandomProjection implements FeatureTransform {
+    private RealVector metricVector;
+    private RealVector transformedVector;
+    private int n;
+    private int k;
+    private RealVector mean;
+    private RealVector covV;
+    private DiagonalMatrix covM;
+    private MultivariateNormalDistribution mnd;
+    private RealMatrix randomProjectionMatrix;
+
+    private final MBStream<Datum> output = new MBStream<>();
+
+    public RandomProjection(MacroBaseConf conf){
+        this.k = conf.getInt(MacroBaseConf.RANDOM_PROJECTION_K, MacroBaseDefaults.RANDOM_PROJECTION_K);
+    }
+
+    @Override
+    public void consume(List<Datum> records) throws Exception {
+        n = records.get(0).getMetrics().getDimension();
+        // TODO: should come from conf
+        mean = new ArrayRealVector(n);
+        covV = new ArrayRealVector(n, 1d/n);
+        covM = new DiagonalMatrix(covV.toArray());
+        mnd = new MultivariateNormalDistribution(mean.toArray(), covM.getData());
```

Comment:
Do we want to have a new random projection every time we call `consume`? Subsequent calls won't be comparable, right?